tests/src/test/scala/it/unibo/scafi/test/functional/stdlib/TestBlockT.scala

```diff
@@ -0,0 +1,166 @@
+package it.unibo.scafi.test.functional.stdlib
+
+import it.unibo.scafi.test.FunctionalTestIncarnation._
+import it.unibo.scafi.test.functional.ScafiAssertions.assertNetworkValues
+import it.unibo.scafi.test.functional.ScafiTestUtils
+import it.unibo.utils.StatisticsUtils._
+import org.scalatest._
+
+
+/*
+Not yet tested:
+- cyclicTimerWithDecay
+- clock
+- impulsesEvery
+ */
+class TestBlockT extends FlatSpec{
+  import ScafiTestUtils._
+
+  val Block_T = new ItWord
+
+  private[this] trait SimulationContextFixture {
+    val net: Network with SimulatorOps = ScafiTestUtils.standardNetwork()
+  }
+
+  private[this] trait TestProgram extends AggregateProgram with StandardSensors with BuildingBlocks
+
+  def unitaryDecay: Int => Int = _ - 1
+  def halving: Int => Int = _ / 2
+
+  Block_T should "support T with unitary decay and 0 floor value" in new SimulationContextFixture {
+    exec(new TestProgram {
+      override def main(): Int = T(10, 0, unitaryDecay)
+    }, ntimes = someRounds)(net)
+
+    assertNetworkValues((0 to 8).zip(List(
+      0, 0, 0,
+      0, 0, 0,
+      0, 0, 0
+    )).toMap)(net)
+  }
+
+  Block_T should "T should initialize as specified" in new SimulationContextFixture {
+    exec(new TestProgram {
+      override def main(): Int = T(10, 0, identity[Int])
+    }, ntimes = fewRounds)(net)
+
+    assertNetworkValues((0 to 8).zip(List(
+      10, 10, 10,
+      10, 10, 10,
+      10, 10, 10
+    )).toMap)(net)
+  }
+
+  Block_T should "T should be consistent in intermediate tests" in new SimulationContextFixture {
```

Comment:
Nice approach! :+1: 
But why do you use `manyManyRounds` as the ceiling? Just use a custom positive int local value `ceiling`.

Reply:
I wanted to make it clear that the value is greater that the number of rounds. The idea behind this test is to be sure that T won't be able to complete, in this way the assertion is performed in an intermediate state